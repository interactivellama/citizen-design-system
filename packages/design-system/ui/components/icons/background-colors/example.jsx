// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import SvgIcon from '../../../shared/svg-icon';

//////////////////////////////////////////////
// State Constructor(s)
//////////////////////////////////////////////

let CaseIcon = props =>
<span className="slds-icon_container slds-icon-standard-case" title="description of icon when needed">
  <SvgIcon className="slds-icon" sprite="standard" symbol="case" />
  <span className="slds-assistive-text">Description of icon</span>
</span>;

let LogACallIcon = props =>
<span className="slds-icon_container slds-icon-standard-log-a-call" title="description of icon when needed">
  <SvgIcon className="slds-icon" sprite="standard" symbol="log_a_call" />
  <span className="slds-assistive-text">Description of icon</span>
</span>;

let CustomIcon = props =>
<span className="slds-icon_container slds-icon-custom-custom25" title="description of icon when needed">
  <SvgIcon className="slds-icon" sprite="custom" symbol="custom25" />
  <span className="slds-assistive-text">Description of icon</span>
</span>;

//////////////////////////////////////////////
// Export
//////////////////////////////////////////////

export let states = [
  {
    id: 'caseicon',
    label: 'Case',
    element: <CaseIcon />
  },
  {
    id: 'logacallicon',
    label: 'Log a Call',
    element: <LogACallIcon />
  },
  {
    id: 'customicon',
    label: 'Custom Icon 25',
    element: <CustomIcon />
  }
];
