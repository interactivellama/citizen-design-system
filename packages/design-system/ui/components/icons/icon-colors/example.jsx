// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import SvgIcon from '../../../shared/svg-icon';

/// ///////////////////////////////////////////
// State Constructor(s)
/// ///////////////////////////////////////////

let Demo = props =>
  <div className="demo-only" {...props}>
    {props.children}
  </div>;

let BaseIcon = props =>
<Demo style={{ padding: '0.5rem', background: '#16325c' }}>
  <span className="slds-icon_container" title="description of icon when needed">
    <SvgIcon className="slds-icon" sprite="standard" symbol="lead" />
    <span className="slds-assistive-text">Description of icon</span>
  </span>
</Demo>;

let DefaultIcon = props =>
<span className="slds-icon_container" title="description of icon when needed">
  <SvgIcon className="slds-icon slds-icon-text-default" sprite="utility" symbol="lock" />
  <span className="slds-assistive-text">Description of icon</span>
</span>;

let WarningIcon = props =>
<span className="slds-icon_container" title="description of icon when needed">
  <SvgIcon className="slds-icon slds-icon-text-warning" sprite="utility" symbol="warning" />
  <span className="slds-assistive-text">Description of icon</span>
</span>;

let ErrorIcon = props =>
<span className="slds-icon_container" title="description of icon when needed">
  <SvgIcon className="slds-icon slds-icon-text-error" sprite="utility" symbol="warning" />
  <span className="slds-assistive-text">Description of icon</span>
</span>;

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

export let states = [
  {
    id: 'baseicon',
    label: 'Base',
    element: <BaseIcon />
  },
  {
    id: 'defaulticon',
    label: 'Default',
    element: <DefaultIcon />
  },
  {
    id: 'warningicon',
    label: 'Warning',
    element: <WarningIcon />
  },
  {
    id: 'erroricon',
    label: 'Error',
    element: <ErrorIcon />
  }
];
