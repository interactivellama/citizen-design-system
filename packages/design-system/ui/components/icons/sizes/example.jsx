// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import SvgIcon from '../../../shared/svg-icon';

//////////////////////////////////////////////
// State Constructor(s)
//////////////////////////////////////////////

let Default = props =>
<span className="slds-icon_container slds-icon-standard-case" title="description of icon when needed">
  <SvgIcon className={`slds-icon${props.size ? ' slds-icon--' + props.size : ''}`} sprite="standard" symbol="case" />
  <span className="slds-assistive-text">Case icon</span>
</span>;

let XSmall = props =>
<span className="slds-icon_container" title="description of icon when needed">
  <SvgIcon className="slds-icon slds-icon-text-warning slds-icon--x-small" sprite="utility" symbol="warning" />
  <span className="slds-assistive-text">Warning Icon</span>
</span>;

//////////////////////////////////////////////
// Export
//////////////////////////////////////////////

export let states = [
  {
    id: 'x-small',
    label: 'Extra Small',
    description: 'Extra-small - (`.{{cssPrefix}}icon--x-small`) - is typically used for small alert icons, with no background color.',
    element: <XSmall />
  },
  {
    id: 'small',
    label: 'Small',
    description: 'Small - (`.{{cssPrefix}}icon--small`) - 1.5rem&times;1.5rem (for icons with a background color).',
    element: <Default size="small" />
  },
  {
    id: 'medium',
    label: 'Medium (default)',
    description: 'Medium - (this default size requires no additional class) - 2rem&times;2rem.',
    element: <Default />
  },
  {
    id: 'large',
    label: 'Large',
    description: 'Large - (`.{{cssPrefix}}icon--large`) - 3rem&times;3rem.',
    element: <Default size="large" />
  }
];
