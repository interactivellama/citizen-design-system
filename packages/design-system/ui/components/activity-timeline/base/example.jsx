// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { ButtonIcon } from 'ui/components/button-icons/base/example';
import { ActivityTimelineItem as Event } from 'ui/components/activity-timeline/event/example';
import { ActivityTimelineItem as Call } from 'ui/components/activity-timeline/call/example';
import { ActivityTimelineItem as Email } from 'ui/components/activity-timeline/email/example';
import { ActivityTimelineItem as Task } from 'ui/components/activity-timeline/task/example';
import SvgIcon from 'app_modules/ui/svg-icon';
import classNames from 'classnames';

export let TimelineIcon = props =>
  <div
    className={classNames(
      'slds-icon_container',
      props.symbol == 'log_a_call' ?
        'slds-icon-standard-log-a-call'
      : 'slds-icon-standard-' + props.symbol
    )}
    title={props.symbol == 'log_a_call' ? 'call' : props.symbol }
  >
    <SvgIcon
      className="slds-icon slds-icon--small"
      sprite="standard"
      symbol={ props.symbol }
    />
  </div>;

export let TimelineActions = props =>
  <div className="slds-timeline__actions">
    <p className="slds-timeline__date">{ props.date || 'Feb 24' }</p>
    <ButtonIcon
      className="slds-button--icon-border-filled slds-button--icon-x-small"
      symbol="down"
      assistiveText={ props.assistiveText || 'More Options' }
      aria-haspopup="true"
      title={ props.title || 'More Options' }
    />
  </div>;

export default (
  <ul className="slds-timeline">
    <li>
      <Task />
    </li>
    <li>
      <Event />
    </li>
    <li>
      <Call />
    </li>
    <li>
      <Email />
    </li>
  </ul>
);
