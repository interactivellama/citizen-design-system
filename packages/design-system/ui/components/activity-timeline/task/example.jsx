// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import MediaObject from 'ui/utilities/media-objects/index.react';
import { Checkbox } from 'ui/components/checkbox/base/example';
import { TimelineIcon, TimelineActions } from 'ui/components/activity-timeline/base/example';

export let ActivityTimelineItem = props =>
  <div className="slds-timeline__item">
    <span className="slds-assistive-text">Task</span>
    <MediaObject
      figureRight={
        <TimelineActions
          title="More Options for Task, Review proposals"
          assistiveText="More Options for Task, Review proposals"
        />
      }
    >
      <MediaObject
        className="slds-timeline__media slds-timeline__media--task"
        figureLeft={ <TimelineIcon symbol="task" /> }
        figureLeftClassName="slds-timeline__icon"
      >
        <MediaObject
          title="Review proposals for EBC deck with larger team and have marketing review this"
          figureLeft={ <Checkbox hideLabel label="mark-complete" /> }
        >
          <h3 className="slds-truncate" title="Review proposals for EBC deck with larger team and have marketing review this">
            <a href="javascript:void(0);">Review proposals for EBC deck with larger team and have marketing review this</a>
          </h3>
          <ul className="slds-list--horizontal slds-wrap">
            <li className="slds-m-right--large">
              <span className="slds-text-title">Contact:</span>
              <span className="slds-text-body--small"><a href="javascript:void(0);">Lei Chan</a></span>
            </li>
            <li className="slds-m-right--large">
              <span className="slds-text-title">Assigned to:</span>
              <span className="slds-text-body--small"><a href="javascript:void(0);">Betty Mason</a></span>
            </li>
          </ul>
        </MediaObject>
      </MediaObject>
    </MediaObject>
  </div>;

export default (
  <ActivityTimelineItem />
);
