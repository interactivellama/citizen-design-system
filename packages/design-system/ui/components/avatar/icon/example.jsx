// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import SvgIcon from 'app_modules/ui/svg-icon';
import { StandardIcon } from 'ui/components/icons/standard/example';
import { Avatar } from 'ui/components/avatar/base/example';
import classNames from 'classnames';


//////////////////////////////////////////////
// Export
//////////////////////////////////////////////

export default (
  <Avatar>
    <StandardIcon />
  </Avatar>
);

export let states = [
  {
    id: 'avatar-square-icon',
    label: 'Square Icon Fallback',
    element:
      <Avatar className="slds-avatar--medium">
        <StandardIcon />
      </Avatar>
  },
  {
    id: 'avatar-circle-icon',
    label: 'Circle Icon Fallback',
    element:
      <Avatar className="slds-avatar--circle slds-avatar--medium">
        <StandardIcon symbol="user" />
      </Avatar>
  },
  {
    id: 'size-x-small',
    label: 'Size - X Small',
    element:
      <Avatar className="slds-avatar--x-small">
        <StandardIcon />
      </Avatar>
  },
  {
    id: 'size-small',
    label: 'Size - Small',
    element:
      <Avatar className="slds-avatar--small">
        <StandardIcon />
      </Avatar>
  },
  {
    id: 'size-medium',
    label: 'Size - Medium',
    element:
      <Avatar className="slds-avatar--medium">
        <StandardIcon />
      </Avatar>
  },
  {
    id: 'size-large',
    label: 'Size - Large',
    element:
      <Avatar className="slds-avatar--large">
        <StandardIcon />
      </Avatar>
  }
];
