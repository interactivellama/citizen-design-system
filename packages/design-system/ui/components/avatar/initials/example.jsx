// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { Avatar } from '../base/example';
import { AvatarInitials } from '../';

export default (
  <Avatar>
    <AvatarInitials
      iconClassName="slds-icon-standard-account"
      title="company name"
      initials="Ac"
    />
  </Avatar>
);

export let examples = [
  {
    id: 'linked-avatar-initials',
    label: 'Wrapped in a link',
    element: (
      <a href="javascript:void(0);">
        <Avatar>
          <AvatarInitials
            iconClassName="slds-icon-standard-account"
            title="company name"
            initials="Ac"
          />
        </Avatar>
      </a>
    )
  }
];
