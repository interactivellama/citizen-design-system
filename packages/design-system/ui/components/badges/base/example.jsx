// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { UtilityIcon } from '../../icons/base/example';
import { Badge, InverseBadge, LightestBadge } from '../';

export default <Badge>Badge Label</Badge>;

export let examples = [
  {
    id: 'inverse',
    label: 'Inverse',
    element: <InverseBadge>Badge Label</InverseBadge>
  },
  {
    id: 'light-with-icon',
    label: 'Light with Icon',
    element: (
      <LightestBadge>
        <UtilityIcon
          containerClassName="slds-m-right_xx-small"
          className="slds-icon_xx-small slds-icon-text-default"
          assistiveText={false}
          symbol="moneybag"
        />
        423 Credits Available
      </LightestBadge>
    )
  },
  {
    id: 'badge-with-child',
    label: 'With Child Element',
    element: (
      <Badge>
        Component In: <strong>Lightning</strong>
      </Badge>
    )
  }
];
