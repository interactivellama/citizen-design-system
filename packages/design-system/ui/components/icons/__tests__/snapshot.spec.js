// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import { ActionIcon } from '../action/example';
const { matchesMarkupAndStyle } = createHelpers(__dirname);

describe('Icon Action', () => {
  it('renders an x-small action icon', () =>
    matchesMarkupAndStyle(<ActionIcon className="slds-icon_x-small" />));
  it('renders an xx-small action icon', () =>
    matchesMarkupAndStyle(<ActionIcon className="slds-icon_xx-small" />));
});
