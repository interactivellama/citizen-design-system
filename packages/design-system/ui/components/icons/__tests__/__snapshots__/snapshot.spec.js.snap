// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Action renders an x-small action icon 1`] = `
<span
  className="slds-icon_container slds-icon_container_circle slds-icon-action-description"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_x-small"
  >
    <use
      xlinkHref="/assets/icons/action-sprite/svg/symbols.svg#description"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;

exports[`Icon Action renders an xx-small action icon 1`] = `
<span
  className="slds-icon_container slds-icon_container_circle slds-icon-action-description"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_xx-small"
  >
    <use
      xlinkHref="/assets/icons/action-sprite/svg/symbols.svg#description"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;

exports[`Utility Icon should render a utility icon 1`] = `
<span
  className="slds-icon_container slds-icon-utility-announcement"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#announcement"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;

exports[`Utility Icon should render a utility icon with currentColor 1`] = `
<span
  className="slds-icon_container slds-icon-utility-announcement slds-current-color"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#announcement"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;

exports[`Utility Icon should render a utility large icon 1`] = `
<span
  className="slds-icon_container slds-icon-utility-announcement"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_large"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#announcement"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;

exports[`Utility Icon should render a utility small icon 1`] = `
<span
  className="slds-icon_container slds-icon-utility-announcement"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_small"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#announcement"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;

exports[`Utility Icon should render a utility x-small icon 1`] = `
<span
  className="slds-icon_container slds-icon-utility-announcement"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_x-small"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#announcement"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;

exports[`Utility Icon should render a utility xx-small icon 1`] = `
<span
  className="slds-icon_container slds-icon-utility-announcement"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_xx-small"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#announcement"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;
